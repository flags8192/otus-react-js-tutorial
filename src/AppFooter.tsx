import React, { PureComponent } from 'react';
import styled from '@emotion/styled';

const Root = styled.div`
  height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

interface IAppFooter {
  countUsers: number;
}
export class AppFooter extends PureComponent<IAppFooter> {
  render() {
    return (
      <Root>
        <p role="userCount">Количество пользователей: {this.props.countUsers}</p>
      </Root>
    );
  }
}
