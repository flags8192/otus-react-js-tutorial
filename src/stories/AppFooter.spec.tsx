import React from 'react';
import { render, screen } from '@testing-library/react';
import { AppFooter } from '@src/AppFooter';

describe('AppBody', () => {
  it('Render <AppFooter> without data', () => {
    const { asFragment } = render(<AppFooter countUsers={50} />);
    expect(asFragment()).toMatchSnapshot();
    const dataMessage = screen.getByText(/Количество пользователей: 50/gi);
    expect(dataMessage).toBeInTheDocument();
  });
});
