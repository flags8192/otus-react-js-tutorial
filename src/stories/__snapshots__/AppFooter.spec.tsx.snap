// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody Render <AppFooter> without data 1`] = `
<DocumentFragment>
  <div
    class="css-1co368a"
  >
    <p
      role="userCount"
    >
      Количество пользователей: 50
    </p>
  </div>
</DocumentFragment>
`;
